packages/ripple-nuxt-tide/lib/module.js

```diff
@@ -2,12 +2,17 @@ import defaults from './config/defaults'
 import * as configLoader from './core/config-loader'
 import { RPL_HEADER } from './config/constants'
 import tideSearchApiMiddleware from '@dpc-sdp/ripple-tide-search-api'
+import logger from './core/logger'
 
 const path = require('path')
 
 const nuxtTide = function (moduleOptions) {
   const options = Object.assign(defaults, this.options.tide, moduleOptions)
 
+  if (!options || (options && !options.baseUrl)) {
+    logger.error('Base URL is not defined', { label: 'NuxtTide' })
+    throw new Error('Base URL is not defined')
+  }
```

Comment:
@tim-yao not sure what are other mandatory data that I need to be checking. Is auth user/password required to have value?

Reply:
@rubyllamadora There is another one `options.site`, but that one should be validated inside site module, not here.